examples/views.py

```diff
@@ -16,18 +24,51 @@ class Index(generic.ListView):
     context_object_name = 'books'
     template_name = 'index.html'
 
+    def get_queryset(self):
+        qs = super().get_queryset()
+        if 'type' in self.request.GET:
+            qs = qs.filter(book_type=int(self.request.GET['type']))
+        return qs
+
+
+class SimpleFormView(SuccessMessageMixin, BSModalFormView):
+    template_name = 'examples/simple.html'
+    form_class = SimpleModalForm
+    success_message = 'Success: your comment \'%(comment)s\' was taken into account!'
+    success_url = reverse_lazy('index')
+
+
+class BookFilterView(BSModalFormView):
+    template_name = 'examples/filter_book.html'
+    form_class = BookFilterForm
+
+    def form_valid(self, form):
+        if "clear" in self.request.POST:
+            # the user has clicked on the 'Clear' button
+            self.filter = ''
+        else:
+            # the user has filtered the list of books
+            self.filter = f'?type={form.cleaned_data["type"]}'
+
+        # call the base form_valid (that will call the get_success_url)
+        resp = super().form_valid(form)
+        return resp
```

Comment:
Please rename to `response`. I'm quite obsessed with readability of the code ;)

Reply:
done